compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -1749,6 +1749,9 @@ error.mismatched.visibility.qualifiers.in.object.field=\
 error.invalid.inclusion.of.object.with.private.members=\
   invalid object type inclusion with an object that has private fields or methods
 
+error.invalid.optional.field.in.record.binding.pattern=\
+  invalid record binding pattern; optional fields of records are not allowed in binding patterns
```

Comment:
shouldn't we be using the term `mapping binding pattern`  instead of `record binging pattern`? since there is no such binding pattern called `record binging pattern`? @MaryamZi WDYT?

Reply:
Yeah, let's use mapping binding pattern. 

We initially had record and tuple binding patterns, before they were allowed with maps and arrays also. That's why there's usage of the terms record and tuple binding patterns.